src/Apps/Order2/Routes/Checkout/Components/Order2FulfillmentDetailsStep/Order2PickupForm.tsx

```diff
@@ -0,0 +1,213 @@
+import {
+  Button,
+  Column,
+  GridColumns,
+  PhoneInput,
+  Spacer,
+  Text,
+} from "@artsy/palette"
+import { validateAndExtractOrderResponse } from "Apps/Order/Components/ExpressCheckout/Util/mutationHandling"
+import { useSetOrderFulfillmentOptionMutation } from "Apps/Order2/Routes/Checkout/Mutations/useSetOrderFulfillmentOptionMutation"
+import { useSetOrderPickupDetailsMutation } from "Apps/Order2/Routes/Checkout/Mutations/useSetOrderPickupDetailsMutation"
+import { richPhoneValidators } from "Components/Address/utils"
+import { countries as phoneCountryOptions } from "Utils/countries"
+import createLogger from "Utils/logger"
+import type { Order2PickupForm_order$key } from "__generated__/Order2PickupForm_order.graphql"
+import { Formik, type FormikHelpers, type FormikValues } from "formik"
+import { useCallback } from "react"
+import { graphql, useFragment } from "react-relay"
+import * as Yup from "yup"
+
+const logger = createLogger("Order2PickupForm.tsx")
+
+interface Order2PickupFormProps {
+  order: Order2PickupForm_order$key
+}
+
+interface PickupFormValues {
+  phoneNumber: string
+  phoneNumberCountryCode: string
+}
+
+export const Order2PickupForm: React.FC<Order2PickupFormProps> = ({
+  order,
+}) => {
+  const orderData = useFragment(FRAGMENT, order)
+  const fulfillmentOptions = orderData?.fulfillmentOptions
+  // By the time we get here, this option should be available
+  const pickupFulfillmentOption = fulfillmentOptions?.find(
+    option => option.type === "PICKUP",
+  )
+
+  const setOrderPickupDetails = useSetOrderPickupDetailsMutation()
+  const setOrderFulfillmentOption = useSetOrderFulfillmentOptionMutation()
+
+  const handleSubmit = useCallback(
+    async (
+      values: FormikValues,
+      formikHelpers: FormikHelpers<PickupFormValues>,
+    ): Promise<void> => {
+      if (!pickupFulfillmentOption) {
+        logger.error(
+          "No pickup fulfillment option available (should not happen)",
+        )
+        return
+      }
+
+      try {
+        const setOrderFulfillmentOptionResult =
+          await setOrderFulfillmentOption.submitMutation({
+            variables: {
+              input: {
+                id: orderData.internalID,
+                fulfillmentOption: pickupFulfillmentOption as {
+                  type: "PICKUP"
+                },
+              },
+            },
+          })
+        validateAndExtractOrderResponse(
+          setOrderFulfillmentOptionResult.setOrderFulfillmentOption
+            ?.orderOrError,
+        )
```

Comment:
re-using this code from express checkout which converts a graphql order mutation into the success type (or throws)